test/Test/Pos/BinarySpec.hs

```diff
@@ -29,22 +38,51 @@ spec = describe "Bi" $ do
             binaryTest @(B.FixedSizeInt Int)
             binaryTest @(B.FixedSizeInt Int64)
         describe "TinyVarInt" $ do
-            -- TODO: check that these actually take 1 byte
             describe "1 byte" $ do
+                it "serializing 0 takes 1 byte" $ serializeFail 0 1
                 prop "0" $ binaryEncodeDecode (B.TinyVarInt 0)
+                it "serializing 1 takes 1 byte" $ serializeFail 1 1
                 prop "1" $ binaryEncodeDecode (B.TinyVarInt 1)
+                it "serializing 0x75 takes 1 byte" $ serializeFail 0x75 1
                 prop "0x75" $ binaryEncodeDecode (B.TinyVarInt 0x75)
+                it "serializing 0x79 takes 1 byte" $ serializeFail 0x79 1
                 prop "0x79" $ binaryEncodeDecode (B.TinyVarInt 0x79)
-            -- TODO: check that these actually take 2 bytes
             describe "2 bytes" $ do
+                it "serializing 0x80 takes 2 bytes" $ serializeFail 0x80 2
                 prop "0x80" $ binaryEncodeDecode (B.TinyVarInt 0x80)
+                it "serializing 0x81 takes 2 bytes" $ serializeFail 0x81 2
                 prop "0x81" $ binaryEncodeDecode (B.TinyVarInt 0x81)
+                it "serializing 0x82 takes 2 bytes" $ serializeFail 0x82 2
                 prop "0x82" $ binaryEncodeDecode (B.TinyVarInt 0x82)
+                it "serializing 0x100 takes 2 bytes" $ serializeFail 0x100 2
                 prop "0x100" $ binaryEncodeDecode (B.TinyVarInt 0x100)
+                it "serializing 0x1000 takes 2 bytes" $ serializeFail 0x1000 2
                 prop "0x1000" $ binaryEncodeDecode (B.TinyVarInt 0x1000)
                 let maxnum = 2^(14::Int)-1
+                it "serializing (2^14)-1 takes 2 bytes" $
+                    B.biSize (B.TinyVarInt maxnum) `shouldBe` 2
```

Comment:
Why not `serializeFail` here?

Also, why is `serializeFail` called like it's called? The name seems non-intuitive to me.